### PR TITLE
Add exporter and integration test for docker

### PR DESCRIPTION
**Pull request by rossf7**
Towards #1 

This adds the exporter code, a Dockerfile and a first integration test that uses the Docker container.

As you can see it took a few commits 😅 but the github action is done. It uses a local docker registry. The integration test is just a go test that connects to the container on port 8000.

Later I'd like to add deployment manifests for Nomad and Kubernetes and we can add more integration tests for them.

### CODE REVIEW

File: .github/workflows/docker-integration-test.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+name: docker-integration-test
+on: [push]
+jobs:
+  docker-integration-test:
+    runs-on: ubuntu-latest
+    services:
+      registry:
+        image: registry:2
+        ports:
+          - 5000:5000
```
Comment (rossf7) [inline .github/workflows/docker-integration-test.yaml:10]: The integration test builds the docker image and pushes it to this local registry.

File: .github/workflows/docker-integration-test.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        with:
+          context: .
+          push: true
+          tags: localhost:5000/grid-intensity-exporter:integration-test
+      -
+        name: Docker run
+        run: docker run -d -p 8000:8000 localhost:5000/grid-intensity-exporter:integration-test
+      - 
+        name: Run integration test
+        run: go test -v -tags=dockerrequired ./integration/test/docker
```
Comment (rossf7) [inline .github/workflows/docker-integration-test.yaml:38]: The integration test is a regular go test. The `-tags` is so it doesn't run if you call `go test ./...`

File: integration/test/docker/metric_test.go [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+		t.Fatalf("expected status code %d: got %d", http.StatusOK, resp.StatusCode)
+	}
+
+	respBytes, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		t.Fatalf("expected nil got %v", err)
+	}
+
+	metrics := string(respBytes)
+	expectedMetricText := "grid_intensity_carbon_actual{provider=\"carbonintensity.org.uk\",region=\"UK\"}"
```
Comment (rossf7) [inline integration/test/docker/metric_test.go:29]: The integration test is pretty basic. It just checks this text is present.

**Comment by mrchrisadams** [comment]
@ofpiyush I'm pretty happy with this, personally. Because there's not much to merge into, are you cool with raising any issue in an separate PR, or making any separate issue for stuff that leaps out at you?

**Comment by mrchrisadams** [comment]
@rossf7 we don't have any real guidelines about merges vs rebases in git repos at TGWF. What do you prefer to work with on go projects?

Once I know, I'll either merge in, or rebase accordingly, as I'm happy with this 👍

**Comment by rossf7** [comment]
> We don't have any real guidelines about merges vs rebases in git repos at TGWF. What do you prefer to work with on go projects?

Thanks @mrchrisadams I'm used to doing squash merges but the history was messy. So I cleaned it up and did a force push. I hope that's OK? I try to avoid force pushes but seemed like a good option in this case.

I don't mind if we merge or rebase. Also I think you can run github actions locally. I'll try that next time as it also means there is a shorter feedback loop when working on the tests.

**Comment by ofpiyush** [comment]
Yes I will find proper time to look at this over the weekend.

No reason to block both of you till then.
I'll create issues if I see something odd.

Squash merge is my default too!

**Comment by rossf7** [comment]
Great, thanks @ofpiyush!

**Pull request merged**